Crash: calling undeclared function with record field of invalid type

```c
struct S {
    does_not_exist x;
};

void foo(void) {
    struct S s = {0};
    bar(s.x);
}
```

```
panic: reached unreachable code
/Users/ehaas/source/arocc/src/aro/TypeStore.zig:167:25: 0x10453ddaf in type (arocc)
            .invalid => unreachable,
                        ^
/Users/ehaas/source/arocc/src/aro/TypeStore.zig:385:38: 0x1044958db in base (arocc)
        while (true) switch (cur.type(comp)) {
                                     ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:8548:78: 0x1047fbb8f in callExpr (arocc)
            if (call_expr.shouldPromoteVarArg(arg_count)) switch (arg.qt.base(p.comp).type) {
                                                                             ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:8250:42: 0x1047f0533 in suffixExpr (arocc)
        .l_paren => return try p.callExpr(lhs),
                                         ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:8168:36: 0x1047f9063 in unExpr (arocc)
            while (try p.suffixExpr(lhs)) |suffix| {
                                   ^
```

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -8785,7 +8785,7 @@ fn callExpr(p: *Parser, lhs: Result) Error!Result {
         if (call_expr.shouldPerformLvalConversion(arg_count)) {
             try arg.lvalConversion(p, param_tok);
         }
-        if (arg.qt.hasIncompleteSize(p.comp) and !arg.qt.is(p.comp, .void)) return error.ParsingFailed;
+        if ((arg.qt.hasIncompleteSize(p.comp) and !arg.qt.is(p.comp, .void)) or arg.qt.isInvalid()) return error.ParsingFailed;
 
         if (arg_count >= params_len) {
             if (call_expr.shouldPromoteVarArg(arg_count)) switch (arg.qt.base(p.comp).type) {

--- a/test/cases/call undeclared function with invalid type.c
+++ b/test/cases/call undeclared function with invalid type.c
@@ -1,0 +1,12 @@
+struct S {
+    does_not_exist x;
+};
+
+void foo(void) {
+    struct S s = {0};
+    bar(s.x);
+}
+
+#define EXPECTED_ERRORS "call undeclared function with invalid type.c:2:5: error: unknown type name 'does_not_exist'" \
+    "call undeclared function with invalid type.c:7:5: error: call to undeclared function 'bar'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]"
+

